ADR/0011-roles-and-permissions.md

```diff
@@ -0,0 +1,81 @@
+# 11. Roles and Permissions for StoneSoup
+
+Date: 2023-01-10
+
+## Status
+
+Accepted
+
+## Context
+StoneSoup is using Kubernetes as the control plane for managing its resources. We require a system for managing user roles and permissions in this context. We have defined the following roles for our project: "Contributor", "Maintainer", and "Owner". We need to map these roles to specific permissions in the Kubernetes RBAC system, in terms of API groups, verbs, and resources.
+
+## Decision
+
+We will use the built-in Kubernetes RBAC system for StoneSoup's role and permissions management, and map the following roles to specific permissions, as described in the table below:
+
+### Roles
+Contributor: Members who interact with the workspace mostly through pull requests.
+Maintainer: Members who manage the workspace without access to sensitive or destructive actions.
+Owner: Members who have full access to the workspace including sensitive and destruction actions.
+
+### Roles and Permissions Table
+
+|     Role      | Permissions             | API Groups                | Verbs                             | Resources
```

Comment:
I would swap Resources & Verbs columns so it's: API Group (not API Group**S**!), Resources, Verbs.

And I also like `Permissions / Roles for` as it's in https://docs.google.com/document/d/1KMwHbvTAJ-AVKdHo-gVq_LkgpmwKKdCtCiI04X69kII/edit#heading=h.wsjpdkxenjjv more than just `Permissions` which is a bit ambiguous.

But it's all cosmetic :)